x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/InstanceTypeService.java

```diff
@@ -13,36 +13,51 @@
 import org.elasticsearch.xcontent.XContentType;
 
 import java.io.IOException;
-import java.io.UncheckedIOException;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.function.Function;
 import java.util.zip.GZIPInputStream;
 
 public class InstanceTypeService {
-    private final Map<InstanceType, CostEntry> costsPerDatacenter = new HashMap<>();
-
-    public void load() {
-        try (
-            GZIPInputStream in = new GZIPInputStream(
-                InstanceTypeService.class.getClassLoader().getResourceAsStream("profiling-costs.json.gz")
-            )
-        ) {
-            XContentParser parser = XContentType.JSON.xContent().createParser(XContentParserConfiguration.EMPTY, in);
-            if (parser.currentToken() == null) {
-                parser.nextToken();
-            }
-            List<Map<String, Object>> rawData = XContentParserUtils.parseList(parser, XContentParser::map);
-            for (Map<String, Object> entry : rawData) {
-                costsPerDatacenter.put(InstanceType.fromCostSource(entry), CostEntry.fromSource(entry));
-            }
 
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
+    private InstanceTypeService() {}
```

Comment:
Nit: While we're at it, can you please declare this class (`InstanceTypeService`) as `final`?